src/octopeer-github/main/Options/Options.ts

```diff
@@ -15,48 +15,55 @@ const Options = new (class Options {
-    public get DNW_COMMENT_ELEMENTS() {return "doNotWatchCommentElements"; }
-    public get DNW_KEYBOARD_EVENTS() {return "doNotWatchKeyboardShortcutEvents"; }
-
-    // A map that contains all option names and their default (boolean) values.
-    private optionMap: { [key: string]: boolean; } = {
-        [this.LOGGING]: true,
-        [this.TRACK_TABS]: true,
-        [this.TRACK_COMMENTS]: true,
-        [this.TRACK_PEER_COMMENTS]: true,
-        [this.TRACK_FOCUS]: true,
-        [this.HASH_USERNAME]: true,
-        [this.HASH_REPO]: true,
-        [this.HASH_FILE]: false,
-        [this.DNW_ON_SCREEN_EVENTS]: true,
-        [this.DNW_HOVER_EVENTS]: true,
-        [this.DNW_COMMENT_ELEMENTS]: true,
-        [this.DNW_KEYBOARD_EVENTS]: true,
+    public get MOUSE_HOVER() {return "mouseHover"; }
+    public get MOUSE_CLICK() {return "mouseClick"; }
+    public get MOUSE_SCROLLING() {return "mouseScrolling"; }
+    public get MOUSE_POSITION() {return "mousePosition"; }
+    public get DATA_COMMENTS() {return "dataComments"; }
+    public get DATA_KEYSTROKES() {return "dataKeystrokes"; }
+    public get DATA_HTML() {return "dataHTML"; }
+    public get DATA_TABS() {return "dataTabs"; }
+    public get DATA_RESOLUTION() {return "dataResolution"; }
+
+    // A map that contains all option names and their (boolean) values.
+    // The first boolean value represents the current value of an option and is updated.
+    // The second boolean value represents the default value of an option.
+    private optionMap: { [key: string]: [boolean]; } = {
+        [this.LOGGING]: [true, true],
+        [this.MOUSE_HOVER]: [true, true],
+        [this.MOUSE_CLICK]: [true, true],
+        [this.MOUSE_SCROLLING]: [true, true],
+        [this.MOUSE_POSITION]: [true, true],
+        [this.DATA_COMMENTS]: [true, true],
+        [this.DATA_KEYSTROKES]: [true, true],
+        [this.DATA_HTML]: [false, false],
+        [this.DATA_TABS]: [true, true],
+        [this.DATA_RESOLUTION]: [true, true],
     };
 
     private observers: OptionsObserver[];
 
     /**
      * Initialization fetches the current settings and stores them in this class.
+     * Dynamic options are set to the sync storage and default options are set to the local storage.
      * Besides, enables a listener that listens for changes in the sync storage area.
      * This means that any items that was changed (newValue) is set if changed;
      */
     public init() {
         this.observers = [];
-        let options: string[] = this.generateOptionList();
+        chrome.storage.sync.get("loggingEnabled", (res) => {
+            let options = this.generateOptionList();
+            if (res[this.LOGGING] === undefined) {
+                chrome.storage.sync.set(this.generateCurrentOptionMap());
+            } else {
+                chrome.storage.sync.get(options, (obj) => this.syncOptionMap(obj));
+            }
+            chrome.storage.local.set(this.generateDefaultOptionMap());
```

Comment:
For functionality purposes. Default options can be stored in the local storage, as on reload of the extension, they won't change. It's nice to keep them separate for this reason.